air-quality-backend/src/etl/in_situ/openaq_adapter.py

```diff
@@ -1,7 +1,37 @@
 from datetime import datetime
 import logging
+from src.etl.air_quality_index.calculator import (
+    get_pollutant_index_level,
+    get_overall_aqi_level,
+)
+from src.etl.air_quality_index.pollutant_type import PollutantType
 from src.etl.in_situ.sort_in_situ import sort_by_distance_and_time
 
+required_pollutant_data = {
+    "o3": PollutantType.OZONE,
+    "no2": PollutantType.NITROGEN_DIOXIDE,
+    "so2": PollutantType.SULPHUR_DIOXIDE,
+    "pm10": PollutantType.PARTICULATE_MATTER_10,
+    "pm25": PollutantType.PARTICULATE_MATTER_2_5,
+}
+
+
+def _extract_pollutants(data):
+    output = []
+    for pollutant in required_pollutant_data.keys():
+        output.append(data[pollutant]["aqi_level"])
+    return output
+
+
+def _calculate_overall_aqi_value(formatted_dataset):
+    for i in range(0, len(formatted_dataset)):
+        # 3 being the amount of non measurement fields in the data dictionary
+        if 3 + len(required_pollutant_data.keys()) == len(formatted_dataset[i].keys()):
```

Comment:
This is a bit unintuitive and could be fragile in the future if we ever wanted to change the document structure.

Can we change it to an explicit check of keys e.g.
```python
all(item in required_pollutant_data.keys() for item in formatted_dataset[i].keys())
```

It would probably be more efficient with sets, but this will do

Reply:
Turns out it goes the other way around: if all(item in formatted_dataset[i].keys() for item in required_pollutant_data.keys()):